### DISCUSSION TITLE
New value for a bool config field

### DISCUSSION THREAD

**Discussion by tmzane** [golangci/golangci-lint#4512] [Q&A] [answered]
Hey team 👋

Just wanted to ask your opinion.

I'm going to add a new value for the `context-only` option in `sloglint` (see https://github.com/go-simpler/sloglint/issues/29 for details). The problem is that this option is already implemented as boolean. I see two options here: add a new option for this specific case or change the type to `any` to permit both `bool` and `string`. Given that users interact with the config only via `yaml` file, no type safety would be lost from this perspective. I don't see usage of `any` fields in the current `golangci-lint` config, so I wonder if there is a preference for such cases in the project.

Thanks.

**Comment by ldez** [comment, marked as answer]
Hello,

I can see several possibilities, one of them is to remove `context-only` and create a new `context`.
The new `context` can be a string with values empty, `all`, `scope1`, `scope2`, etc.

Inside golangci-lint we can create a deprecation for `context-only`:
- if `false` we ignore the option
- if `true` we log a warning and check if there is a value inside `context`, if not we fall back on `all`.
